### ISSUE TITLE
.../libgit2/src/repository.c:1168: git_repository_index__weakptr: Assertion `out && repo' failed

### ISSUE THREAD

**Issue by mgood7123** [libgit2/libgit2#4708]
You are opening a _bug report_ against the libgit2 project: we use
GitHub Issues for tracking bug reports and feature requests.  If you
have a question about an API or usage, please ask on StackOverflow:
http://stackoverflow.com/questions/tagged/libgit2.  If you want to
have high-level discussions about the libgit2 project itself, visit
https://github.com/libgit2/discussions.

Otherwise, to report a bug, please fill out the reproduction steps
(below) and delete these introductory paragraphs.  Thanks!

### Reproduction steps
curl https://paste.pound-python.org/raw/qNo0HXbwEzjbxZajxDVf/ -o lib.c ; gcc lib.c --std=gnu99 -o libg -lgit2 ; ./libg
### Expected behavior
cabox@box-codeanywhere:~/workspace$ ./libg                                                              
testing
argc = 2
argv[0] = init
argv[1] = test
Initialized empty Git repository in ~/CCR/UserFiles/Mobile C/Git/git/test/
argc = 3
argv[0] = add
argv[1] = -v
argv[2] = ./test
testing done                                                              
cabox@box-codeanywhere:~/workspace$
### Actual behavior
cabox@box-codeanywhere:~/workspace$ ./libg                                                              
testing                                                                                                
argc = 2                                                                                                
argv[0] = init                                                                                          
argv[1] = test                                                                                          
Initialized empty Git repository in /home/cabox/workspace/test/                                        
argc = 3                                                                                                
argv[0] = add                                                                                          
argv[1] = -v                                                                                            
argv[2] = ./test                                                                                        
libg: /home/cabox/workspace/libgit2/src/repository.c:1168: git_repository_index__weakptr: Assertion `out
&& repo' failed.                                                                                      
Could not open repositoryAborted                                                                        
cabox@box-codeanywhere:~/workspace$
### Version of libgit2 (release number or SHA1)
2.7
### Operating system(s) tested
debian

**Comment by ethomson** [comment]
> seems to assert when opening a non existing repo

Indeed.  Your error handling code only prints "Could not open repository" but continues.  It then tries to get the index of the repository, calling `git_repository_index` with a `NULL` repository pointer, which we `assert` must be non-null.

**Comment by mgood7123** [comment]
Ok
